packages/plugin/src/__tests__/compile.test.ts

```diff
@@ -1,5 +1,48 @@
-import { test } from 'vitest';
+import { expect, test } from "vitest";
+import { compile } from "../compile.js";
+import {
+  MOCK_EXPORTS_WITH_PROMISE,
+  MOCK_JSON_SAFE_EXPORTS,
+  MOCK_JSON_UNSAFE_EXPORTS,
+} from "../__mocks__/compile.mock.js";
 
-test.todo('generates default export');
-test.todo('generates named exports');
-test.todo('generates resolved promises');
+test("generates default export", async () => {
+  const code = await compile(MOCK_JSON_SAFE_EXPORTS);
+  expect(code).toContain(
```

Comment:
This is a good case for a snapshot test (https://vitest.dev/guide/snapshot.html#snapshot).

Reply:
Duplicate of my comment in discord:

Why would a snapshot test be better than just using toContain?

Say in the future the compile changes to somehow reorder the exports, wouldn't toMatchInlineSnapshot fail then?

Or is that a _good fail_, as in, we'd know that "oh yea x y and z gets affected by the change, and we'll just update the snapshot"?